src/SFA.DAS.EAS.Portal.Client/Database/Models/AccountDocument.cs

```diff
@@ -11,20 +11,16 @@ public class AccountDocument : Document
 
         [JsonProperty("accountId")]
         public long AccountId => Account.Id;
-        private AccountDocument(long accountId) : base(1)
+        public AccountDocument(long accountId) : base(1)
```

Comment:
Not sure why this has changed In this story but the factory pattern may be preferable to protect from future changes/creation rules.  Raise as a tech debt issue if you think it should be changed.  Please stick to the story/use case.

Reply:
If there a need for a factory method at a later date, we can introduce one. At the moment, I can't see any value in having one.

Reply:
To encapsulate the setting of the IsNew flag.

Reply:
The `IsNew` flag is always set to `true` when constructed through the public constructor. (I've made the `JsonConstructor` attributed constructor private, as it provided a back door to constructing an `AccountDocument` without setting up invariants.)

The construction of `AccountDocument` is encapsulated in the constructor. We don't need to encapsulate setting individual properties that the constructor sets.